libkbfs/bserver_remote.go

```diff
@@ -405,7 +405,10 @@ func (b *BlockServerRemote) Get(ctx context.Context, tlfID tlf.ID, id kbfsblock.
 				id, tlfID, context, size)
 			dbc := b.config.DiskBlockCache()
 			if dbc != nil {
-				go dbc.Put(ctx, tlfID, id, buf, serverHalf)
+				// This used to be called in a goroutine to prevent blocking
+				// the `Get`. But we need this cached synchronously for
+				// later behavior.
```

Comment:
Please expand "later behavior".  Presumably something to do with the prefetch logic?

Reply:
If the block doesn't appear in the disk block cache when we try anything in `PutInCaches`, we can't `UpdateMetadata` on the block, since it's not yet there. Also, if we try to `Put` in a goroutine, then caching/prefetching operations become racy since the state of the metadata when the goroutine was initiated may be different from what the Prefetcher is trying to do, for example. So we get some really ugly undefined behavior.